src/test/java/ru/svgogin/service/spark/web/SparkControllerItTest.java

```diff
@@ -0,0 +1,234 @@
+            + "    \"statusDate\": \"2020-11-30\"\n"
+            + "  },\n"
+            +
+            "    "
+            + "{\n"
+            + "        \"inn\": \"9705113553\",\n"
+            + "        \"ogrn\": \"5177746290288\",\n"
+            + "        \"kpp\": \"772501001\",\n"
+            + "        \"fullNameRus\": \"ОБЩЕСТВО С ОГРАНИЧЕННОЙ ОТВЕТСТВЕННОСТЬЮ \\\"СЕВЕНТЕК\\\"\",\n"
+            + "        \"shortNameRus\": \"ООО \\\"7ТЕК\\\"\",\n"
+            + "        \"statusName\": \"Actual\",\n"
+            + "        \"statusDate\": \"2021-01-30\"\n"
+            + "    }\n"
+            + "    ]\n"
+            )
+        );
+  }
+
+  @Test
+  void getCompanyByInnShouldReturnCompany() throws Exception {
+    // given
+    aggregateTemplate.insert(bank);
+    // when
+    mockMvc.perform(MockMvcRequestBuilders.get("/spark/companies/07725038124"))
+        .andDo(print())
+        .andExpect(MockMvcResultMatchers.status().isOk())
+        .andExpect(MockMvcResultMatchers.content().json(
+            "{\n"
+            + "    \"inn\": \"07725038124\",\n"
+            + "    \"ogrn\": \"01037739527077\",\n"
+            + "    \"kpp\": \"0770401001\",\n"
+            + "    \"fullNameRus\": \"АКЦИОНЕРНОЕ ОБЩЕСТВО \\\"БАНК ДОМ.РФ\\\"\",\n"
+            + "    \"shortNameRus\": \"АО\\\"БАНК ДОМ.РФ\\\"\",\n"
+            + "    \"statusName\": \"Действующая\",\n"
+            + "    \"statusDate\": \"2020-11-30\"\n"
+            + "  }\n"
+            )
+        );
+  }
+  @Test
+  void saveCompanyShouldReturnCompanyWithCorrectArgs() throws Exception {
+    // given
+    // when
+    mockMvc.perform(MockMvcRequestBuilders.post("/spark/companies/")
+        .content("{\n"
+                 + "    \"inn\": \"07725038124\",\n"
+                 + "    \"ogrn\": \"01037739527077\",\n"
+                 + "    \"kpp\": \"0770401001\",\n"
+                 + "    \"fullNameRus\": \"АКЦИОНЕРНОЕ ОБЩЕСТВО \\\"БАНК ДОМ.РФ\\\"\",\n"
+                 + "    \"shortNameRus\": \"АО\\\"БАНК ДОМ.РФ\\\"\",\n"
+                 + "    \"statusName\": \"Действующая\",\n"
+                 + "    \"statusDate\": \"2020-11-30\"\n"
+                 + "  }\n").contentType("application/json"))
+        .andDo(print())
+        .andExpect(MockMvcResultMatchers.status().isOk())
+        .andExpect(MockMvcResultMatchers.content().json(
+            "{\n"
+            + "    \"inn\": \"07725038124\",\n"
+            + "    \"ogrn\": \"01037739527077\",\n"
+            + "    \"kpp\": \"0770401001\",\n"
+            + "    \"fullNameRus\": \"АКЦИОНЕРНОЕ ОБЩЕСТВО \\\"БАНК ДОМ.РФ\\\"\",\n"
+            + "    \"shortNameRus\": \"АО\\\"БАНК ДОМ.РФ\\\"\",\n"
+            + "    \"statusName\": \"Действующая\",\n"
+            + "    \"statusDate\": \"2020-11-30\"\n"
+            + "  }\n"
+            )
+        );
+
+    // then
+    var result = sparkRepositoryDb.findByInn("07725038124").orElse(test7tec); //What is better to throw if company was not found?
```

Comment:
yes, you should throw exception or use optional and that value is present.

Reply:
Exception added